app/models/competition.rb

```diff
@@ -2661,6 +2664,18 @@ def disconnect_all_payment_integrations
     competition_payment_integrations.destroy_all
   end
 
+  def can_change_registration_system?
+    return false unless registration_not_yet_opened?
+
+    if uses_new_registration_service?
+      return true
+    elsif self.registrations.any?
+      return false
+    else
+      return true
```

Comment:
As an exercise, try merging this into one boolean expression with `&&` and/or `||` operators 😉

Reply:
Took it down to 3 separate booleans - I like the readability/clarity of not combining statements which are logically separate but can change if you prefer to compound them